+adi/+AD4170/Rx.m

```diff
@@ -1,120 +1,26 @@
-classdef Rx < adi.common.Rx & matlabshared.libiio.base & adi.common.Attribute
+classdef Rx < adi.common.Rx & adi.common.RxTx ...
+        & adi.AD4170.Base
     % AD4170 Precision ADC Class
     % 
     % adi.AD4170.Rx Receives data from the AD4170 ADC
     % The adi.AD4170.Rx System object is a signal source that can receive
     % data from the AD4170.
     %
     %   `rx = adi.AD4170.Rx;`
-    %   `rx = adi.AD4170.Rx('serial:COM18,230400');`
+    %   `rx = adi.AD4170.Rx('uri','ip:192.168.2.1');`
     %
-    % `AD4170 Datasheet <https://www.analog.com/media/en/technical-documentation/data-sheets/ad4170.pdf>`_
-
-    properties (Nontunable)
-        % SampleRate Sample Rate
-        %   Baseband sampling rate in Hz, specified as a scalar
-        %   in samples per second.
-        SampleRate = '32500'
-
-        % SamplesPerFrame Samples Per Frame
-        %   Number of samples per frame, specified as an even positive
-        %   integer.
-        SamplesPerFrame = 1024
-
-        %AdcMode ADC Mode
-        % ADC Mode for data capture, represented as a string.
-        % It supports the following: 'Continuous_Conversion',
-        % 'Continuous_Conversion_FIR', 'Continuous_Conversion_IIR',
-        % 'Standby', 'Power_Down', 'Idle'
-        AdcMode = 'Continuous_Conversion'
-
-        %Fs Fs
-        % Fs for configuring the sampling rate of the device.
-        % It supports a numeric value. Please refer to the DS
-        % for more details on configuring FS
-        Fs = 20
-
-    end
-
-    % Channel names
-    properties (Nontunable, Hidden, Constant)
-        channel_names = { ...
-                         'voltage0', 'voltage1', 'voltage2', 'voltage3'}
-    end
-
-    % isOutput
-    properties (Hidden, Nontunable, Access = protected)
-        isOutput = false
-    end
+    % `AD4170 Datasheet <https://www.analog.com/media/en/technical-documentation/data-sheets/ad4170-4.pdf>`_
 
     properties (Nontunable, Hidden)
-        Timeout = Inf
-        kernelBuffersCount = 1
-        dataTypeStr = 'int32'
-        phyDevName = 'ad4170'
-        devName = 'ad4170'
-    end
-
-    properties (Nontunable, Hidden, Constant)
-        Type = 'Rx'
-    end
-
-    properties (Hidden, Constant)
-        ComplexData = false
+        channel_names = {'voltage0','voltage1','voltage2'...
+            'voltage3','voltage4'};
```

Comment:
In the base class, we've setters till channel 3 only. Should we also add ch4 there?

Reply:
I'd limit it to channel3- as the default configurations in the FW are only till ch4